### PR TITLE
Timeline playback

### PR DESCRIPTION
**Pull request by j8seangel**
This PR is the first step for the timeline playback feature.
- Includes UI buttons
- Set the speed and play and pause functionality

This approach will need review and brainstorming as it isn't performing well at all when `10x` speed is selected.

### CODE REVIEW

**Comment by nerik** [comment]
@j8seangel Can you please review my changes ? Also, I didn't touch the CSS but it seems there's an issue introduced in this branch.
Thanks 🐠

File: src/timebar/timebar.js [hunk 1]
```diff
@@ -29,7 +28,15 @@ const ONE_DAY_MS = 1000 * 60 * 60 * 24
...
     super()
+    const toggleImmediate = (immediate) => {
+      this.setState((state) => ({
+        immediate,
+      }))
+    }
+    this.interval = null
     this.state = {
+      immediate: false,
+      toggleImmediate,
```
Comment (j8seangel) [inline src/timebar/timebar.js]: do we need to have the toggleImmediate inside the state? I think is not a good pattern and it isn't necessary

**Comment by j8seangel** [comment]
Styles issues were addresses, feel free to merge once you review the comments. Thanks!

File: src/timebar/components/playback.js [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+        this.togglePlay(false)
+      }
+    }
+    if (clamped !== 'end' || loop === true) {
+      return true
+    }
+  }
+
+  tick = (elapsedMs) => {
+    if (this.lastUpdateMs === undefined || this.lastUpdateMs === null) {
```
[the remark this answers is not in the document]
Comment (nerik) [reply src/timebar/components/playback.js]: Very cool, didn't think of that 👍

File: src/timebar/components/playback.js [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+
+  onBackwardClick = () => {
+    this.update(-FAST_FORWARD_REWIND_MULTIPLICATOR)
+  }
+
+  onSpeedClick = () => {
+    const { speedStep } = this.state
+    const nextStep = speedStep === SPEED_STEPS.length - 1 ? 0 : speedStep + 1
+    const step = this.getStep(nextStep)
+    this.step = step
```
[the remark this answers is not in the document]
Comment (nerik) [reply src/timebar/components/playback.js]: Thanks, fixed

File: src/timebar/timebar.js (hunk 1 above)
Comment (nerik) [reply src/timebar/timebar.js]: I agree and I fixed it. Although I'm wondering: why do you think React docs encourage doing that? : https://reactjs.org/docs/context.html#updating-context-from-a-nested-component (see app.js)

**Comment by nerik** [comment]
@j8seangel Thanks for the review, fixed all. I'm still seeing issues with CSS:

Screenshot 2019-05-15 at 10 51 58

File: src/timebar/timebar.js (hunk 1 above)
Comment (j8seangel) [reply src/timebar/timebar.js]: I guess because they pass the entire state to the context... thanks for the change!

**Comment by j8seangel** [comment]
I couldn't replicate the buttons CSS issue in either chrome, safari or firefox so merging for now and will keep an eye on that.

**Pull request merged**